user_tools/src/spark_rapids_tools/storagelib/cspfs.py

```diff
@@ -99,6 +99,20 @@ def __init__(self, *args: Any, **kwargs: Any):
     def create_as_path(self, entry_path: Union[str, BoundedCspPath]) -> BoundedCspPath:
         return self._path_meta.path_class(entry_path=entry_path, fs_obj=self)
 
+    @classmethod
+    def copy_file(cls, src: BoundedCspPath, dest: BoundedCspPath):
+        """
+        Copy a single file between FileSystems. This function assumes that
```

Comment:
any particular filesystems that are supported or restrictions?

Reply:
No, any FS that is registered in our implementation should work fine. this includes the following list:
```
@register_fs_class("adls", "PyFileSystem")
class AdlsFs(CspFs):...

@register_fs_class('gcs', 'GcsFileSystem')
class GcsFs(CspFs): ...

@register_fs_class("hdfs", "HadoopFileSystem")
class HdfsFs(CspFs): ...

@register_fs_class('local', 'LocalFileSystem')
class LocalFs(CspFs):

@register_fs_class('s3', 'S3FileSystem')
class S3Fs(CspFs):
```